commons/src/test/java/io/aiven/kafka/tieredstorage/commons/storage/s3/S3StorageConfigTest.java

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright 2023 Aiven Oy
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.aiven.kafka.tieredstorage.commons.storage.s3;
+
+import java.util.Map;
+
+import org.apache.kafka.common.config.ConfigException;
+
+import com.amazonaws.auth.AWSCredentialsProvider;
+import com.amazonaws.auth.AWSStaticCredentialsProvider;
+import com.amazonaws.auth.EnvironmentVariableCredentialsProvider;
+import com.amazonaws.regions.Regions;
+import com.amazonaws.services.s3.AmazonS3;
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+class S3StorageConfigTest {
+
+    // Test scenarios
+    // - Minimal/Full configs
+    @Test
+    void minimalConfig() {
+        final String bucketName = "b1";
+        final Map<String, Object> configs = Map.of(S3StorageConfig.S3_BUCKET_NAME_CONFIG, bucketName);
```

Comment:
This could be argued about, but I think we should prefer string literals over named constants in the tests. Thus, we increase the chance of catching inconsistencies, typos, and various type of config cumbersomeness.